src/runtime/typemanager.cs

```diff
@@ -422,6 +434,198 @@ internal static IntPtr BasicSubType(string name, IntPtr base_, Type impl)
+            internal static int sq_contains = 41;
+            internal static int sq_inplace_concat = 42;
+            internal static int sq_inplace_repeat = 43;
+            internal static int sq_item = 44;
+            internal static int sq_length = 45;
+            internal static int sq_repeat = 46;
+            internal static int tp_alloc = 47;
+            internal static int tp_base = 48;
+            internal static int tp_bases = 49;
+            internal static int tp_call = 50;
+            internal static int tp_clear = 51;
+            internal static int tp_dealloc = 52;
+            internal static int tp_del = 53;
+            internal static int tp_descr_get = 54;
+            internal static int tp_descr_set = 55;
+            internal static int tp_doc = 56;
+            internal static int tp_getattr = 57;
+            internal static int tp_getattro = 58;
+            internal static int tp_hash = 59;
+            internal static int tp_init = 60;
+            internal static int tp_is_gc = 61;
+            internal static int tp_iter = 62;
+            internal static int tp_iternext = 63;
+            internal static int tp_methods = 64;
+            internal static int tp_new = 65;
+            internal static int tp_repr = 66;
+            internal static int tp_richcompare = 67;
+            internal static int tp_setattr = 68;
+            internal static int tp_setattro = 69;
+            internal static int tp_str = 70;
+            internal static int tp_traverse = 71;
+            internal static int tp_members = 72;
+            internal static int tp_getset = 73;
+            internal static int tp_free = 74;
+            internal static int nb_matrix_multiply = 75;
+            internal static int nb_inplace_matrix_multiply = 76;
+            internal static int am_await = 77;
+            internal static int am_aiter = 78;
+            internal static int am_anext = 79;
+            internal static int tp_finalize = 80;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct PY_TYPE_SLOT
+        {
+            public long slot; //slot id, from typeslots.h
+            public IntPtr func; //function pointer of the function implementing the slot
+        }
+
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+        internal class PyTypeSpecOffset
+        {
+            static PyTypeSpecOffset()
+            {
+                Type type = typeof(PyTypeSpecOffset);
+                FieldInfo[] fi = type.GetFields();
+                int size = IntPtr.Size;
+                for (int i = 0; i < fi.Length; i++)
+                {
+                    fi[i].SetValue(null, i * size + TypeOffset.ob_size);
+                }
+            }
+
+            public static IntPtr AllocPyTypeSpec(string typename, int obSize, int obFlags, IntPtr slotsPtr)
+            {
+                byte[] ascii = System.Text.Encoding.ASCII.GetBytes(typename);
```

Comment:
.NET supports Unicode type names. Not sure about Python here.

Reply:
@lostmsu I think its possible.  Maybe we should use UTF8 here - I should add a test to see for sure